plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/ProjectContextProvider.kt

```diff
@@ -310,7 +310,7 @@ class ProjectContextProvider(val project: Project, private val encoderServer: En
         chunksMap.forEach { (filePath, chunkList) ->
             var text = ""
             chunkList.forEach { chunk -> text += (chunk.context ?: chunk.content) }
-            val document = RelevantDocument(filePath, text)
+            val document = RelevantDocument(filePath, text.take(10200))
```

Comment:
10240 characters.

Reply:
also double check field programming language

Reply:
The LSP restricts the length of 10k characters, not sure why it can exceed for JetBrains.

Reply:
synced offline, the reason why it surpass the limit is because of the consolidating of chunks at file level, we are removing the consolidation